src/main/webapp/app/exam/participate/exercises/modeling/modeling-exam-submission.component.ts

```diff
@@ -58,20 +57,7 @@ export class ModelingExamSubmissionComponent extends ExamSubmissionComponent imp
      * Checks whether there are pending changes in the current model. Returns true if there are unsaved changes, false otherwise.
      */
     public hasUnsavedChanges(): boolean {
-        // TODO: we should have a callback by Apollon. Whenever a student changes anything in the model, this will immediately set
-        // this.studentSubmission.isSynced = false, then we can simply write here: return !this.studentSubmission.isSynced!;
-        if (!this.modelingEditor || !this.modelingEditor.isApollonEditorMounted) {
-            return false;
-        }
-        const currentApollonModel = this.modelingEditor.getCurrentModel();
-
-        if (this.studentSubmission && this.studentSubmission.model) {
-            const currentSubmissionModel = JSON.parse(this.studentSubmission.model);
-            const versionMatch = currentSubmissionModel.version === currentApollonModel.version;
-            const modelMatch = stringifyIgnoringFields(currentSubmissionModel, 'size') === stringifyIgnoringFields(currentApollonModel, 'size');
-            return versionMatch && !modelMatch;
-        }
-        return false;
+        return !this.studentSubmission.isSynced!;
```

Comment:
Why is the second ! needed here? Makes it look really strange :D

Reply:
I made it similar to other exercise components